cfme/tests/storage/test_volume.py

```diff
@@ -20,19 +20,35 @@
 
 
 @pytest.mark.uncollectif(lambda provider: not provider.one_of(OpenStackProvider))
-def test_volume_navigation(openstack_provider):
+def test_volume_navigation(openstack_provider, appliance):
     # grab a volume name, the table returns a generator so use next
-    view = navigate_to(Volume, 'All')
+
+    collection = VolumeCollection(appliance=appliance)
+    view = navigate_to(collection, 'All')
+
     try:
-        volume_name = view.entities.table[0].name.text
-    except (StopIteration, NoSuchElementException):
+        volume_name = view.entities.get_first_entity().name
+    except(StopIteration, ItemNotFound):
         pytest.skip('Skipping volume navigation for details, no volumes present')
-    volume = Volume(name=volume_name, provider=openstack_provider)
+
+    volume = collection.instantiate(name=volume_name, provider=openstack_provider)
 
     assert view.is_displayed
 
     view = navigate_to(volume, 'Details')
     assert view.is_displayed
 
-    view = navigate_to(Volume, 'Add')
+    view = navigate_to(collection, 'Add')
     assert view.is_displayed
+
+
+@pytest.mark.uncollectif(lambda provider: not provider.one_of(OpenStackProvider))
+def test_volume_collective_crud(openstack_provider, appliance):
+    collection = VolumeCollection(appliance=appliance)
```

Comment:
OPTIONAL: You can create a fixture for the collection in cfme/fixtures/model_collections.py. There it even injects the appliance fixture as well.